src/js/components/ArticleForm.js

```diff
@@ -12,19 +12,37 @@ export default React.createClass({
     return {title: "", url: ""};
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    this.setState({
+      title: nextProps.current && nextProps.current.title || "",
+      url:   nextProps.current && nextProps.current.url || ""
```

Comment:
Yeah, matter of taste; but agreed this is too much boilerplate for what it brings, will update.